fedimint-core/src/module/mod.rs

```diff
@@ -964,6 +971,59 @@ impl<T: Encodable + Decodable + 'static> SerdeModuleEncoding<T> {
     }
 }
 
+impl<T> fmt::Debug for SerdeModuleEncoding2<T>
+where
+    T: Encodable + Decodable,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str("SerdeModuleEncoding2(")?;
+        fmt::Debug::fmt(&AbbreviateHexBytes(&self.0), f)?;
+        f.write_str(")")?;
+        Ok(())
+    }
+}
+
+impl<T: Encodable + Decodable> From<&T> for SerdeModuleEncoding2<T> {
+    fn from(value: &T) -> Self {
+        let mut bytes = vec![];
+        fedimint_core::encoding::Encodable::consensus_encode(value, &mut bytes)
+            .expect("Writing to buffer can never fail");
+        Self(bytes, PhantomData)
+    }
+}
+
+impl<T: Encodable + Decodable + 'static> SerdeModuleEncoding2<T> {
+    pub fn try_into_inner(&self, modules: &ModuleDecoderRegistry) -> Result<T, DecodeError> {
+        let mut reader = std::io::Cursor::new(&self.0);
+        Decodable::consensus_decode(&mut reader, modules)
+    }
+
+    /// In cases where we know exactly which module kind we expect but don't
+    /// have access to all decoders this function can be used instead.
+    ///
+    /// Note that it just assumes the decoded module instance id to be valid
+    /// since it cannot validate against the decoder registry. The lack of
+    /// access to a decoder registry also makes decoding structs impossible that
+    /// themselves contain module dyn-types (e.g. a module output containing a
+    /// fedimint transaction).
+    pub fn try_into_inner_known_module_kind(&self, decoder: &Decoder) -> Result<T, DecodeError> {
```

Comment:
Sus. Why.

Reply:
TODO: verify that the original wasn't written by me and silly.

Reply:
OK, so it's used to decode session/output outcomes in client modules sometimes, doesn't panic. We always take it from current consensus of peers (except now, where we check the signature so effectively the same thing). Looks OK.